File: src/main/java/br/ufsc/cultivar/models/Place.java
package br.ufsc.cultivar.models;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.*;
import lombok.experimental.FieldDefaults;
import lombok.experimental.Wither;
import org.hibernate.validator.constraints.br.CNPJ;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;

@Value
@Wither
@Builder
@EqualsAndHashCode(callSuper = true)
@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
@JsonDeserialize(builder = Place.PlaceBuilder.class)
@JsonInclude(JsonInclude.Include.NON_NULL)
@ApiModel("Company/School")
public class Place extends AbstractModel<String>{

    @NotBlank
    @CNPJ
    @ApiModelProperty(name = "CNPJ", notes = "Company/School's CNPJ", required = true)
    String id;
    @NotBlank
    @ApiModelProperty(notes = "Name of company/school", required = true)
    String name;
    @NotBlank
    @ApiModelProperty(notes = "Contact number of company/school", required = true)
    String phone;
    @NotNull
    @ApiModelProperty(notes = "Flag for identify school from companies")
    Boolean school;
    @NotNull
    @ApiModelProperty(notes = "User responsible in the company/school", required = true)
    User responsible;
    @NotNull
    @ApiModelProperty(notes = "Address of company/school", required = true)
    Address address;

    @JsonPOJOBuilder(withPrefix = "")
    public static class PlaceBuilder {
        public PlaceBuilder(){}
    }
}

File: webapp/src/model/place.js
import User from './user';
import Address from './address';
import Roles from './role';
import Status from './status';
export default class {
    constructor(id = '', name = '', phone = '', school = false,
            responsible = new User(), address = new Address()){
        this.id = id;
        this.name = name;
        this.phone = phone;
        this.school = school;
        if(school){
            responsible.role = Roles.SCHOOL_ADMIN;
        } else {
            responsible.role = Roles.COMPANY_ADMIN
        }
        responsible.status = Status.APPROVED
        this.responsible = responsible;
        this.address = address;
    }
}